src/v2/Apps/Order/Components/PollAccountBalance.tsx

```diff
@@ -0,0 +1,140 @@
+import { FC, useState } from "react"
+import { graphql, createRefetchContainer, RelayRefetchProp } from "react-relay"
+import { Spinner } from "@artsy/palette"
+import styled from "styled-components"
+
+import { useSystemContext } from "v2/System"
+import { SystemQueryRenderer } from "v2/System/Relay/SystemQueryRenderer"
+import { renderWithLoadProgress } from "v2/System/Relay/renderWithLoadProgress" // todo
+import { usePoll } from "v2/Utils/Hooks/usePoll"
+import {
+  PollAccountBalanceQuery,
+  PollAccountBalanceQueryResponse,
+} from "v2/__generated__/PollAccountBalanceQuery.graphql"
+import { PollAccountBalance_commerceBankAccountBalance } from "v2/__generated__/PollAccountBalance_commerceBankAccountBalance.graphql"
+
+interface PollAccountBalanceProps {
+  relay: RelayRefetchProp
+  setupIntentId: string
+  commerceBankAccountBalance: PollAccountBalance_commerceBankAccountBalance | null
+  onBalanceCheckComplete: (displayInsufficientFundsError: boolean) => void
+  buyerTotalCents: number
+  currencyCode: string
+}
+
+const PollAccountBalance: FC<PollAccountBalanceProps> = ({
+  relay,
+  setupIntentId,
+  commerceBankAccountBalance,
+  onBalanceCheckComplete,
+  buyerTotalCents,
+  currencyCode,
+}) => {
+  const [shouldStopPolling, setShouldStopPolling] = useState(false)
+  const timeoutID = setTimeout(() => setShouldStopPolling(true), 5000)
+
+  usePoll({
+    callback: () => {
+      relay.refetch({ setupIntentId }, null, {}, { force: true })
+    },
+    intervalTime: 1300,
+    key: setupIntentId,
+    clearWhen: shouldStopPolling,
+  })
+
+  // we receive the balance from Stripe and it's enough to buy the artwork
+  const userHasEnoughBalance =
+    commerceBankAccountBalance?.balanceCents &&
+    commerceBankAccountBalance?.balanceCents >= buyerTotalCents &&
+    currencyCode === commerceBankAccountBalance?.currencyCode
```

Comment:
#non-blocking. I wonder if the variables can be less repetitive (and below) by just using `balanceCents` and `balanceCurrencyCode`.